src/neuroglancer/point_user_layer.ts

```diff
@@ -0,0 +1,129 @@
+ * limitations under the License.
+ */
+
+import {ChunkManager} from 'neuroglancer/chunk_manager/frontend';
+import {CoordinateTransform} from 'neuroglancer/coordinate_transform';
+import {getPoint, GetPointOptions} from 'neuroglancer/datasource/factory';
+import {UserLayer, UserLayerDropdown} from 'neuroglancer/layer';
+import {LayerListSpecification, registerLayerType, registerVolumeLayerType} from 'neuroglancer/layer_specification';
+import {getVolumeWithStatusMessage} from 'neuroglancer/layer_specification';
+import {Overlay} from 'neuroglancer/overlay';
+// import {VolumeType} from 'neuroglancer/sliceview/base';
+// import {FRAGMENT_MAIN_START, getTrackableFragmentMain, ImageRenderLayer} from
+// 'neuroglancer/sliceview/image_renderlayer'; import {SliceViewPanelPointLayer, PointLayer} from
+// 'neuroglancer/point/frontend';
+import {MultiscalePointChunkSource, RenderLayer} from 'neuroglancer/point/frontend';
+import {PointRenderLayer} from 'neuroglancer/point/point_renderlayer';
+import {StatusMessage} from 'neuroglancer/status';
+import {trackableAlphaValue} from 'neuroglancer/trackable_alpha';
+import {mat4} from 'neuroglancer/util/geom';
+import {verifyOptionalString} from 'neuroglancer/util/json';
+import {makeWatchableShaderError} from 'neuroglancer/webgl/dynamic_shader';
+import {RangeWidget} from 'neuroglancer/widget/range';
+import {ShaderCodeWidget} from 'neuroglancer/widget/shader_code_widget';
+
+require('./image_user_layer.css');
+require('neuroglancer/help_button.css');
+require('neuroglancer/maximize_button.css');
+
+function getPointsWithStatusMessage(
+    chunkManager: ChunkManager, x: string,
+    options: GetPointOptions = {}): Promise<MultiscalePointChunkSource> {
+  return StatusMessage.forPromise(
+      new Promise(function(resolve) {
+        resolve(getPoint(chunkManager, x, options));
+      }),
+      {
+        initialMessage: `Retrieving metadata for point source ${x}.`,
+        delay: true,
+        errorPrefix: `Error retrieving metadata for point source ${x}: `,
+      });
+}
+
+export class PointUserLayer extends UserLayer {
+  pointsPath: string|undefined;
+  opacity = trackableAlphaValue(0.5);
+  //   fragmentMain = getTrackableFragmentMain();
+  //   shaderError = makeWatchableShaderError();
+  renderLayer: RenderLayer;
+  constructor(manager: LayerListSpecification, x: any) {
+    super();
+
+    this.opacity.restoreState(x['opacity']);
+
+    let pointsPath = this.pointsPath = verifyOptionalString(x['point']);
+    if (pointsPath !== undefined) {
+      getPointsWithStatusMessage(manager.chunkManager, pointsPath).then(points => {
+        if (!this.wasDisposed) {
+          let renderLayer = this.renderLayer =
+              new PointRenderLayer(points, {opacity: this.opacity, sourceOptions: {}});
+          this.addRenderLayer(renderLayer);
+        }
+      });
+    }
+  }
+  toJSON() {
+    let x: any = {'type': 'point'};
+    x['point'] = this.pointsPath;
```

Comment:
Perhaps make this 'source' rather than 'point' for consistency with other layers.